packages/react/src/Button/ButtonBase.tsx

```diff
@@ -64,45 +84,65 @@ const ButtonBase = forwardRef(
     }
 
     return (
-      <StyledButton
-        as={Component}
-        sx={sxStyles}
-        {...rest}
-        ref={innerRef}
-        data-block={block ? 'block' : null}
-        data-size={size === 'small' || size === 'large' ? size : undefined}
-        data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
-        data-inactive={inactive ? true : undefined}
-      >
-        {Icon ? (
-          <Icon />
-        ) : (
-          <>
-            <Box as="span" data-component="buttonContent" sx={getAlignContentSize(alignContent)}>
-              {LeadingVisual && (
-                <Box as="span" data-component="leadingVisual" sx={{...iconWrapStyles}}>
-                  <LeadingVisual />
+      <ConditionalWrapper if={Boolean(loading)} sx={{display: block ? 'block' : 'inline-block'}} data-loading-wrapper>
+        <StyledButton
+          as={Component}
+          sx={sxStyles}
+          {...rest}
+          ref={innerRef}
+          data-block={block ? 'block' : null}
+          data-inactive={inactive ? true : undefined}
+          data-loading={Boolean(loading)}
+          data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
+          data-size={size === 'small' || size === 'large' ? size : undefined}
+          aria-disabled={loading ? true : undefined}
+          aria-describedby={[loadingAnnouncementID, ariaDescribedBy]
```

Comment:
Is it safe to only render `loadingAnnouncementID` when `loading === true`? Or do we need `loadingAnnouncementID` to be in `aria-describedby` on initial render for screen readers to pick this up?